include/beast/http/basic_headers.hpp

```diff
@@ -244,9 +243,10 @@ class basic_headers_base::const_iterator
     value.
 
     Field names are stored as-is, but comparison are case-insensitive.
-    The container preserves the order of insertion of fields with
-    different names. For fields with the same name, the implementation
-    concatenates values inserted with duplicate names as per rfc7230.
+    When the container is iterated, the fields are presented in the order
+    of insertion. For fields with the same name, the container behaves
+    as a std::multiset; there will be a separate value for each occurrence
```

Comment:
Doesn't it behave more like a `std::list`, at least if iterators are used?  `begin()` and `end()` return a `list_t::iterator`.